quine-cypher/src/main/scala/com/thatdot/quine/compiler/cypher/QueryPart.scala

```diff
@@ -12,29 +12,54 @@ object QueryPart {
     *
     * @param queryPart query to compiler
     * @param isEntireQuery this query part is the whole query
+    * @param isSubQuery is this inside a `CALL { .. }`?
     * @return execution instructions for Quine
     */
   def compile(
     queryPart: ast.QueryPart,
-    isEntireQuery: Boolean = true
+    isEntireQuery: Boolean = true,
+    isSubQuery: Boolean = false
   ): CompM[cypher.Query[cypher.Location.Anywhere]] =
     queryPart match {
-      case ast.SingleQuery(clauses) =>
-        compileClauses(clauses, isEntireQuery)
+      case sq: ast.SingleQuery =>
+        if (!isSubQuery) {
+          compileClauses(sq.clauses, isEntireQuery)
+        } else {
+          for {
+            // Prepare for the subquery to run by setting the imported columns
+            initialColumns: Vector[Symbol] <- CompM.getColumns
+            importedVariables =
+              if (sq.isCorrelated) {
+                sq.importColumns.view.map(Symbol.apply).toVector
+              } else {
+                initialColumns
+              }
+            () <- CompM.clearColumns
```

Comment:
Feels like we can skip this `clearColumns` + `traverse addColumn` in the case where `isCorrelated` is false?